### PR TITLE
Adding default methods

### PR DESCRIPTION
**Pull request by j8r**
I've added default methods for `Router#run` and `Router#close` to allow us not to write them.
I've changed the name `Router#draw_routes` to `Router#routes`, because more concise/declarative to list routes.

The performance seems to be close to the same. There are also some formatting.

Side note: we can use `Struct` instead of `Class` to gain speed in some cases, I've changed the `samples` accordingly.

### CODE REVIEW

File: shard.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name: router
-version: 0.1.13
+version: 0.3.0
```
Comment (tbrand) [inline shard.yml:2]: Why is the version 0.2.0 skipped?

**Review by tbrand — CHANGES_REQUESTED** [review]
I think this is not a good structure since the Router doesn't require users to implement `def routes` at compile level. I mean the Router should be abstract class then. I would like to keep this library as a middleware, not server it self. 

Here is a summery of my opinions.

1. Renaming `draw_routes` to `routes` seems not smart, renaming to `draw` is acceptable.
2. If you want to wrap router.cr, please define another abstract class.
3. Other bit fixes are acceptable.

Anyway, thanks for the pull request! :+1:

File: shard.yml (hunk 1 above)
Comment (j8r) [reply shard.yml:2]: Because of https://github.com/tbrand/router.cr/releases, the version seems to be in reality `0.2.1`

**Comment by j8r** [comment]
I have made an `abstract class` first, but this way we can't inherit to another one anymore. With `include Router`, we are free to inherit of another class.

File: sample/tips.cr [hunk 2]
```diff
@@ -12,58 +12,37 @@ require "../src/router"
...
+#    StaticFileHandler will pass the access to the RouteHandler
+#    if the file or directory does not exist.
+# 4. So RouteHandler should be last.
+#
+# The array of the handlers should be like this.
+# Note: if a route can't be handled by RouteHandler (a.k.a. route not found)
+# and this handler is the last, a 404 error response will be returned;
+# otherwise the execution will continue with the next handler in a row
+
+web_server = WebServer.new.run(handlers: [HTTP::LogHandler.new(STDOUT),
```
Comment (j8r) [inline sample/tips.cr:42]: What do you think @tbrand about having the handlers in the class constructor (`WebServer.new(handlers: [...]).run`)? I've tried to do so but haven't succeed.

**Comment by tbrand** [comment]
Sorry for the late reply.

I'm rethinking about your implementation, but it's still a bit strange.

1. A class can be a server when it includes `Router`.
1. A class have to implement `draw` method. But it's not checked at compiling level.

That's too bad for a usability since user cannot be noticed when compiling a class including Router without `draw` implementation.

And may be you misunderstand current implementation. That's my bad since the document is not good.
The point is that a class including `Router` **do NOT have to have a draw_routes method**.
The **name** of `draw_routes` is not important, you can change it `draw` without modifying current router.cr's implementation.

So if you want to proceed this changes, please separate this into;
1. Just change a `draw_routes` to `draw` at sample codes.
2. Implement an abstract class for wrapping Router. If you want to avoid to implement abstract class, you have to force user to implement a `draw` method at compile level. (But I have no idea without using abstract class.)
3. Other bit patches.

1 and 3 are smoothly acceptable but we still have to discuss about 2.

**Comment by j8r** [comment]
I've already changed to `draw` now.
I've created a new empty `draw` method – the user no longer needs to worry about implementing it or not. There is no error at compile time anymore.

The main question is if we have to set standard method names. The user can implement their own `initialize`, `draw`, `run` or whatever method that will override the default ones .

**Comment by j8r** [comment]
Closing the PR - I realize it has more its place in a web framework that here, that aims not be minimalist. Sorry!

**Pull request closed without merging**